scripts/makebumpver.j2

```diff
@@ -0,0 +1,463 @@
+            m = re.match(branch_pattern, self.git_branch)
+            if m:
+                return m.group(1)
+            rhel_branch_pattern = r"^rhel-(\d+)(.*)"
+            m = re.match(rhel_branch_pattern, self.git_branch)
+            if m:
+                return m.group(1)
+        return False
+
+    def _get_commit_detail(self, commit, field):
+        proc = run_program(['git', 'log', '-1', "--pretty=format:%s" % field, commit])
+        ret = proc[0].strip('\n').split('\n')
+
+        if len(ret) == 1 and ret[0].find('@') != -1:
+            ret = [ret[0].split('@')[0]]
+        elif len(ret) == 1:
+            ret = [ret[0]]
+        else:
+            ret = [x for x in ret if x != '']
+
+        return ret
+
+    def _rpm_log(self, fixedIn):
+        git_range = "%s-%s-%s.." % (self.name, self.version, self.release)
+        proc = run_program(['git', 'log', '--no-merges', '--pretty=oneline', git_range])
+        lines = proc[0].strip('\n').split('\n')
+
+        for commit in self.ignore:
+            lines = [x for x in lines if not x.startswith(commit)]
+
+        rpm_log = []
+        bad = False
+
+        for line in lines:
+            if not line:
+                continue
+            fields = line.split(' ')
+            commit = fields[0]
+
+            summary = self._get_commit_detail(commit, "%s")[0]
+            body = self._get_commit_detail(commit, "%b")
+            author = self._get_commit_detail(commit, "%aE")[0]
+
+            if re.match(r".*(#infra).*", summary) or re.match(r"infra: .*", summary):
+                print("*** Ignoring (#infra) commit %s\n" % commit)
+                continue
+
+            if re.match(r".*(#test).*", summary):
+                print("*** Ignoring (#test) commit %s\n" % commit)
+                continue
+
+            if self.rhel:
+                bad, bugs = self._check_rhel_issues(commit, summary, body, author)
+                rpm_log.append(("%s (%s)" % (summary.strip(), author), list(bugs)))
+            else:
+                rpm_log.append(("%s (%s)" % (summary.strip(), author), None))
+
+        if bad:
+            sys.exit(1)
+
+        return rpm_log
+
+    def _check_rhel_issues(self, commit, summary, body, author):
+        issues = set()
+        bad = False
+
+        summary = summary + f" ({author})"
+
+        for bodyline in body:
+            m = re.match(r"^(Resolves|Related):\ +jira#\w+-\d+.*$", bodyline)
+            if not m:
+                continue
+
+            actionre = re.search("(Resolves|Related)", bodyline)
+            bugre = re.search(r"\w+-\d+", bodyline)
+
+            if actionre and bugre:
+                action = actionre.group()
+                bug = bugre.group()
+
+                # store the bug to output list if checking is disabled and continue
+                if self.skip:
+                    issues.add(f"{action}: jira#{bug}")
+                    print(f"*** Bug {bug} Related commit {commit} is skipped\n")
```

Comment:
Is the commit really skipped? Just the checks are skipped, aren't they?

Reply:
I see, yeah, it might be more clear with `skipping checks` probably.